keras/backend/tensorflow/optimizer_distribute_test.py

```diff
@@ -163,30 +163,24 @@ def test_ema(self):
     def test_gradient_accumulation(self):
         with self.strategy.scope():
             v = backend.Variable([[1.0, 2.0], [3.0, 4.0]])
-            grads = backend.convert_to_tensor([[1.0, 1.0], [1.0, 1.0]])
+            grads = backend.convert_to_tensor([[1.0, 1.0], [2.0, 2.0]])
```

Comment:
Thanks for adding the test, can we test with the grad that has different value on each replica? 

You can create a distribute value via https://www.tensorflow.org/api_docs/python/tf/distribute/DistributedValues

Reply:
Best I can tell, it is not possible to call an optimizer (or any other Keras function) with `DistributeValues`. The reason is that `DistributeValues` is a stand-in for a tensor, but it does not implement the tensor API (no `.shape`, no `.dtype`, etc).